### PR TITLE
fix RibbonTrail coverity issues; CID: 127543, 216057, 216060, 216066

### PR DESCRIPTION
**Pull request by ricab**
Addresses #2246

### CODE REVIEW

File: Source/Urho3D/Graphics/RibbonTrail.cpp [hunk 1]
```diff
@@ -53,13 +53,31 @@ inline bool CompareTails(TrailPoint* lhs, TrailPoint* rhs)
...
+TrailPoint::TrailPoint() :
+    TrailPoint{Vector3::ZERO, Vector3::ZERO}
+{
+}
+
+TrailPoint::TrailPoint(const Vector3& position, const Vector3& forward) :
+    position_{position},
+    forward_{forward},
+    parentPos_{Vector3::ZERO},
+    elapsedLength_{0.0f},
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/RibbonTrail.cpp]: I suggest to move scalar initializers into header, mark `TrailPoint()` as `=default`, and keep only `TrailPoint(const Vector3& position, const Vector3& forward) : position_(...), forward_(...) {}`

Comment (ricab) [reply Source/Urho3D/Graphics/RibbonTrail.cpp]: Ok sure, will do that.

**Comment by ricab** [comment]
@eugeneko is this what you meant?

**Comment by eugeneko** [comment]
Exactly. Thanks.

**Comment by weitjong** [comment]
This PR can be merged as is then. However, it would be great if you can add a period to end the doxygen comment and also at the unrelated LUA script comment that you have touched. Only inline comments do not end with period. That’s current convention. And while you at it, may be also could change the member init in the header using `{}` instead of `= 0` or `= nullptr`.

**Comment by ricab** [comment]
Done. I will try to remember the periods in non-inline comments. BTW, the reason why I changed the lua file was that, while searching for TrailPoint occurrences that could be affected in script, I noticed the wrong comment, and though another PR would be a bit of an overkill for that. But now I noticed none of the other comments in it have periods at the end, so another PR may actually be needed :laughing:

**Comment by weitjong** [comment]
Thanks for that. Yeah, I guess only the doxygen comments are more glaring because they will end up in our online documentation. Also you are right that sometime we also make such mistakes ourselves or may not catch them while reviewing PR every time. But that’s not an excuse.

**Pull request merged**